### PR TITLE
Features/newsfeed

### PR DESCRIPTION
**Pull request by viennguyen24**
**Issue:** #84 

**Branch:** features/newsfeed

**Files changed:** 
1. client\src\components\community\post\ImageSlider.tsx 
2. client\src\components\community\post\index.ts
3. client\src\components\community\post\MockPosts.tsx
4. client\src\components\community\post\Post.tsx
5. client\src\components\community\post\PostHeader.tsx
6. client\src\components\community\post\PostFooter.tsx
7. client\src\pages\Feed.tsx

**Overview**:
This will be the main page to display the newsfeed for our community feature. Right now the page is still using mock data
image

**Demo:**
[demo](https://www.loom.com/share/f68c60e8b5b443c095c371943a43f4b7?sid=e39860dd-5e3f-4a34-abab-de7f2c41274b)
[Also for responsiveness](https://www.loom.com/share/867512cdea5a49f0bfb53265d49db6b5?sid=4a881914-e4ff-46a2-9292-53eb8ad9b6c9)
**Remaining work:**

- [x] Call backend to generate newsfeed when first load
- [x] Update new post for infinite scroll
- [x] Update post when user click like, comment, share
- [x] Find way (if possible) to assure image and video quality
- [x] Allow user to modify post if they are the author
- [ ] Add more to the UI to make it less blank

### CODE REVIEW

File: client/src/pages/Feed.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { MockPosts } from '../components/community/post/MockPosts';
+import { Box } from '@chakra-ui/react';
+
+const style = {
+  backgroundColor: 'gray.50',
+};
+
+const Feed = () => {
+  return (
+    <Box width="100%" height="100%" overflow="auto" backgroundColor="blackAlpha.50" paddingX="30%">
```
Comment (hungngodev) [inline client/src/pages/Feed.tsx]: Great now use react query to link with backend. Populate your mongodb with some fake created post or add in manually

**Review by hungngodev — CHANGES_REQUESTED** [review]

**Review by hungngodev — CHANGES_REQUESTED** [review]
Now please implement the like functions, share functions the comment function will be left for later on when you connect with Vy PR

File: client/src/pages/Feed.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
+import { Box, Image } from '@chakra-ui/react';
+import { useState, useEffect } from 'react';
+import { z } from 'zod';
+import { MockPosts, PostObjectType, PostObject } from '../components/community/post/MockPosts';
+import { Post } from '../components/community/post/index';
+import customFetch from '../utils/customFetch';
+import { useQuery } from '@tanstack/react-query';
+import { Player } from '@lottiefiles/react-lottie-player';
+
```
Comment (hungngodev) [inline client/src/pages/Feed.tsx:17]: You could add another function called loader. Loader is a component of react router dom that is the same use as useEffect with [] as empty dependency array. Usually when using with react query, we use it for ensure the query data meaning that we check if the data for this page is cached or not, if it is cached then no fetch and vice versa. You can check my Meal page.

**Review by hungngodev — CHANGES_REQUESTED** [review]

**Comment by viennguyen24** [comment]
> Now please implement the like functions, share functions the comment function will be left for later on when you connect with Vy PR

Hi Hung I implemented the like function at the back-end route PR first [at this commit](https://github.com/hungngodev/Flavorie/pull/92/commits/e2d3fda7cd78aa300a55f950e605451f10f6905f)

File: client/src/pages/Feed.tsx (hunk 2 above)
Comment (viennguyen24) [reply client/src/pages/Feed.tsx:17]: > You could add another function called loader. Loader is a component of react router dom that is the same use as useEffect with [] as empty dependency array. Usually when using with react query, we use it for ensure the query data meaning that we check if the data for this page is cached or not, if it is cached then no fetch and vice versa. You can check my Meal page.

You can check out the Loader function at [the latest commit](https://github.com/hungngodev/Flavorie/pull/89/commits/c7bd0b0a73d5435f63fe74c3b31740ddb2b18408), I will fix the conflict with the app file once everything is okay

**Review by hungngodev — APPROVED** [review]
Great changes

**Pull request merged**